tifeatures/layer.py

```diff
@@ -304,14 +333,13 @@ async def query(
         dt: str = None,
         limit: Optional[int] = None,
         offset: Optional[int] = None,
+        bbox_only: Optional[bool] = None,
+        simplify: Optional[float] = None,
     ) -> Tuple[FeatureCollection, int]:
         """Build and run Pg query."""
-        if not self.geometry_columns:
-            raise MissingGeometryColumn("Must have geometry column for geojson output.")
 
-        geometry_column = self.geometry_column(geom)
-        if not geometry_column:
-            raise InvalidGeometryColumnName(f"Invalid Geometry Column: {geom}.")
```

Comment:
I think we should still raise an error when `geom` is wrong, except when it's `none`

Reply:
- What if a user wants to filter on a specific geometry_column but do not want geometry back?
- could a User want to filter on a specific geometry column but want another back?

Reply:
Good point. This would still technically be possible with CQL2 since you still have to explicitly use the geometry column name for a CQL2 filter. This would also let you search on one geometry column and return another. Not the most obvious, but it would work. Open to other ideas.